src/IssueViz.Security.UnitTests/Hotspots/LocalHotspotStoreTests.cs

```diff
@@ -287,6 +287,91 @@ public void UpdateForFile_ServerHotspots_SameFileUpdate_MakesUnmatchedServerHots
         serverStoreMock.Verify(store => store.GetAll(), Times.Once);
     }
 
+    [TestMethod]
+    // Regression test for https://github.com/SonarSource/sonarlint-visualstudio/issues/4602
```

Comment:
The regression test was difficult to write and is quite fragile.

The other new test `UpdateForFile_MatchingDoesNotDependOnOrderServerHotspotsAreReturnedByTheServer()` is simpler. It's not an exact repro of the bug, but it also failed before I added the comparer fix.

I'm tempted to delete this regression test and rely on the other new test. What do you think?